File: analytics/src/test/java/io/appmetrica/analytics/impl/utils/validation/ValidationResultTest.java
package io.appmetrica.analytics.impl.utils.validation;

import io.appmetrica.analytics.testutils.CommonTest;
import org.assertj.core.api.SoftAssertions;
import org.junit.Test;

import static org.mockito.Mockito.mock;

public class ValidationResultTest extends CommonTest {

    @Test
    public void testSuccessful() {
        Validator validator = mock(Validator.class);
        ValidationResult result = ValidationResult.successful(validator);
        SoftAssertions softAssertion = new SoftAssertions();
        softAssertion.assertThat(result.isValid()).as("result").isTrue();
        softAssertion.assertThat(result.getDescription()).as("description").isEmpty();
        softAssertion.assertThat(result.getValidatorClass()).as("validatorClass")
            .isSameAs(validator.getClass());
        softAssertion.assertAll();
    }

    @Test
    public void testFailed() {
        Validator validator = mock(Validator.class);
        ValidationResult result = ValidationResult.failed(validator, "error");
        SoftAssertions softAssertion = new SoftAssertions();
        softAssertion.assertThat(result.isValid()).as("result").isFalse();
        softAssertion.assertThat(result.getDescription()).as("description").isEqualTo("error");
        softAssertion.assertThat(result.getValidatorClass()).as("validatorClass")
            .isSameAs(validator.getClass());
        softAssertion.assertAll();
    }

}

File: core-utils/src/test/java/io/appmetrica/analytics/coreutils/internal/validation/ValidationResultTest.kt
package io.appmetrica.analytics.coreutils.internal.validation

import io.appmetrica.gradle.testutils.CommonTest
import org.assertj.core.api.SoftAssertions
import org.junit.Test
import org.mockito.kotlin.mock

class ValidationResultTest : CommonTest() {

    @Test
    fun successful() {
        val validator = mock<Validator<*>>()
        val result = ValidationResult.successful(validator)

        SoftAssertions().apply {
            assertThat(result.isValid).`as`("result").isTrue
            assertThat(result.description).`as`("description").isEmpty()
            assertThat(result.validatorClass).`as`("validatorClass").isSameAs(validator.javaClass)
            assertAll()
        }
    }

    @Test
    fun failed() {
        val validator = mock<Validator<*>>()
        val result = ValidationResult.failed(validator, "error")

        SoftAssertions().apply {
            assertThat(result.isValid).`as`("result").isFalse
            assertThat(result.description).`as`("description").isEqualTo("error")
            assertThat(result.validatorClass).`as`("validatorClass").isSameAs(validator.javaClass)
            assertAll()
        }
    }
}
